domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresConnectionManager.java

```diff
@@ -0,0 +1,96 @@
+package org.folio.rest.persist;
+
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.Pool;
+import io.vertx.core.Future;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.tools.utils.Envs;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PostgresConnectionManager {
+  private static final Logger LOG = LogManager.getLogger(PostgresConnectionManager.class);
+  private static final int DEFAULT_CACHE_SIZE = 4;
+  private final int cacheSizeLimit;
+  private final List<PgConnection> connectionCache = new ArrayList<>();
+  private String currentTenant = "";
+  private int currentIndex;
+
+  public PostgresConnectionManager () {
+    cacheSizeLimit = Envs.getEnv(Envs.DB_MAXSHAREDPOOLSIZE) != null ?
+        Integer.parseInt(Envs.getEnv(Envs.DB_MAXSHAREDPOOLSIZE)) : DEFAULT_CACHE_SIZE;
+  }
+
+  public void clearCache() {
+    currentTenant = "";
+    currentIndex = 0;
+    connectionCache.clear();
+
+    LOG.debug("Cleared connection manager");
+  }
+
+  public Future<PgConnection> getConnection(Pool pool, String schemaName, String tenantId) {
```

Comment:
can we change the datatype of the pool parameter from `Pool` to `PgPool`?

Reply:
Changing it to `Pool` was done because of a sonar warning